payment/controllers_test.go

```diff
@@ -0,0 +1,534 @@
+	suite.Assert().NoError(err)
+	suite.Assert().Equal("paid", updatedOrder.Status)
+
+	// Get all the transactions, should only be one
+
+	handler = GetTransactions(service)
+	req, err = http.NewRequest("GET", "/v1/orders/{orderID}/transactions", nil)
+	rctx = chi.NewRouteContext()
+	rctx.URLParams.Add("orderID", order.ID.String())
+	getReq := req.WithContext(context.WithValue(req.Context(), chi.RouteCtxKey, rctx))
+
+	suite.Require().NoError(err)
+
+	rr = httptest.NewRecorder()
+	handler.ServeHTTP(rr, getReq)
+
+	suite.Assert().Equal(http.StatusOK, rr.Code)
+	var transactions []Transaction
+	err = json.Unmarshal(rr.Body.Bytes(), &transactions)
+	suite.Assert().NoError(err)
+
+	// Check the transaction
+	suite.Assert().Equal(decimal.NewFromFloat32(4.75), transactions[0].Amount)
+	suite.Assert().Equal("uphold", transactions[0].Kind)
+	suite.Assert().Equal("completed", transactions[0].Status)
+	suite.Assert().Equal("BAT", transactions[0].Currency)
+	suite.Assert().Equal(createRequest.ExternalTransactionID, transactions[0].ExternalTransactionID)
+	suite.Assert().Equal(order.ID, transactions[0].OrderID)
+}
+
+func fundWallet(t *testing.T, destWallet *uphold.Wallet, amount decimal.Decimal) {
+	var donorInfo wallet.Info
+	donorInfo.Provider = "uphold"
+	donorInfo.ProviderID = os.Getenv("DONOR_WALLET_CARD_ID")
+	{
+		tmp := altcurrency.BAT
+		donorInfo.AltCurrency = &tmp
+	}
+
+	donorWalletPublicKeyHex := os.Getenv("DONOR_WALLET_PUBLIC_KEY")
+	donorWalletPrivateKeyHex := os.Getenv("DONOR_WALLET_PRIVATE_KEY")
+	var donorPublicKey httpsignature.Ed25519PubKey
+	var donorPrivateKey ed25519.PrivateKey
+	donorPublicKey, err := hex.DecodeString(donorWalletPublicKeyHex)
+	if err != nil {
+		t.Fatal(err)
+	}
+	donorPrivateKey, err = hex.DecodeString(donorWalletPrivateKeyHex)
+	if err != nil {
+		t.Fatal(err)
+	}
+	donorWallet := &uphold.Wallet{Info: donorInfo, PrivKey: donorPrivateKey, PubKey: donorPublicKey}
+
+	if len(donorWallet.ID) > 0 {
+		t.Fatal("FIXME")
+	}
+
+	_, err = donorWallet.Transfer(altcurrency.BAT, altcurrency.BAT.ToProbi(amount), destWallet.Info.ProviderID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	balance, err := destWallet.GetBalance(true)
+	if err != nil {
+		t.Error(err)
+	}
+
+	if balance.TotalProbi.Equals(decimal.Zero) {
+		t.Error("Submit with confirm should result in a balance.")
+	}
+}
+
+func generateWallet(t *testing.T) *uphold.Wallet {
+	var info wallet.Info
+	info.ID = uuid.NewV4().String()
+	info.Provider = "uphold"
+	info.ProviderID = ""
+	{
+		tmp := altcurrency.BAT
+		info.AltCurrency = &tmp
+	}
+
+	publicKey, privateKey, err := httpsignature.GenerateEd25519Key(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	info.PublicKey = hex.EncodeToString(publicKey)
+	newWallet := &uphold.Wallet{Info: info, PrivKey: privateKey, PubKey: publicKey}
+	err = newWallet.Register("bat-go test card")
+	if err != nil {
+		t.Fatal(err)
+	}
+	return newWallet
+}
+
+func (suite *ControllersTestSuite) TestE2E() {
```

Comment:
maybe we should rename this and the other end to end test to more clearly indicate the difference / overlap?

Reply:
https://github.com/brave-intl/bat-go/pull/226/commits/c7f78c25cd9f92a6488ab6da92eeb5a8f3c5c7e1